test/packet/PacketTest.cpp

```diff
@@ -0,0 +1,68 @@
+#include "Packet.hpp"
+#include "configs.hpp"
+
+#include <gtest/gtest.h>
+#include <netinet/in.h>
+#include <vector>
+
+std::vector<uint8_t> make_test_packet(const uint32_t source_ip,
+                                      const uint32_t dest_ip) {
+  // 0100 (IP version) 0101 (IHL Flag)
+  constexpr uint8_t test_packet[] = {0x45, 0x00, 0x00, 0x00, 0x00, 0x00,
+                                     0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+  // Create a vector to hold the full packet (header + IPs)
+  std::vector<uint8_t> packet(test_packet, test_packet + sizeof(test_packet));
+
+  // Swap endianness
+  auto src = ntohl(source_ip);
+  auto dst = ntohl(dest_ip);
```

Comment:
If we're simulating a packet coming from the network we should convert host to network long

```suggestion
  auto src = htonl(source_ip);
  auto dst = htonl(dest_ip);
```

Reply:
updated on new commite